YoCode/HelperMethods.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace YoCode
+{
+    public static class HelperMethods
+    {
+        public static bool ContainsAny(this string line, IEnumerable<string> keywords)
+        {
+            return keywords.Any(line.Contains);
+        }
+
+        public static bool ContainsAll(this string line, IEnumerable<string> keywords)
+        {
+            return keywords.All(line.Contains);
+        }
+
+        public static string GetLineWithAllKeywords(this string output, List<String> keywords)
+        {
+            var sr = new StringReader(output);
+            string line;
+            while ((line = sr.ReadLine()) != null)
+            {
+                if (line.ContainsAll(keywords))
+                {
+                    return line;
+                }
+            }
+            return "";
+        }
```

Comment:
Don't feel like you have to change it but in the spirit of showing what C# can do, this function could be replaced with:
```
return output.Split(Environment.NewLine).FirstOrDefault(line => line.ContainsAll(keywords)) ?? "";
```
It's not quite the same though. According to the documentation, `StringReader.ReadLine()` checks for `"\r"`, `"\n"` and `"\r\n"` whereas `Environment.NewLine` will always only be `"\r\n"` on Windows.

Reply:
It might be worth having unit tests on any of these helper methods that are doing anything complicated, just to make sure it's doing what you expect. I wouldn't bother for `ContainsAny` and `ContainsAll` but maybe `GetLineWithAllKeywords` could have some tests since it's doing a bit of work.

Reply:
Some of the people completing these job tests will be on a non-Windows system, so may be using unix style line endings. So it's worth keeping that in mind when writing these tests.